### DISCUSSION TITLE
Gitea: Large release notes may cause job to fail due to platform errors

### DISCUSSION THREAD

**Discussion by cghislai** [renovatebot/renovate#15281] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:32.24.0

### Please select which platform you are using if self-hosting.

Gitea

### If you're self-hosting Renovate, tell us what version of the platform you run.

gitea/gitea:1.15.10

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

While setting up renovate, after adding a github token to fetch release notes, renovate bot job was failing because of gitea returning an error 500:

```
This PR has been generated by [Renovate Bot](https://github.com/renovatebot/renovate). 0 0 0 1650283464 false 99] - 3.08048ms
--
Mon, Apr 18 2022 2:04:24 pm | 2022/04/18 12:04:24 modules/web/route.go:70:func1() [I] [SQL] ROLLBACK [] - 224.592µs
Mon, Apr 18 2022 2:04:24 pm | 2022/04/18 12:04:24 .../api/v1/repo/pull.go:599:EditPullRequest() [E] UpdateIssueByAPI: Error 1406: Data too long for column 'content' at row 1
Mon, Apr 18 2022 2:04:24 pm | 2022/04/18 12:04:24 Completed PATCH /api/v1/repos/Valuya/embedded-webview/pulls/6 500 Internal Server Error in 40.601052ms

```

Renovate logs below.

No reproduction repository, but i see its caused by a large changelog section in the google-api-nodejs-client changelog: https://github.com/googleapis/google-api-nodejs-client/blob/main/CHANGELOG.md

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

"renovateVersion": "32.24.0"
--
Mon, Apr 18 2022 2:02:55 pm | INFO: Dependency extraction complete (repository=Valuya/vattools)
Mon, Apr 18 2022 2:02:55 pm | "baseBranch": "dev",
Mon, Apr 18 2022 2:02:55 pm | "stats": {
Mon, Apr 18 2022 2:02:55 pm | "managers": {"maven": {"fileCount": 1, "depCount": 10}},
Mon, Apr 18 2022 2:02:55 pm | "total": {"fileCount": 1, "depCount": 10}
Mon, Apr 18 2022 2:02:55 pm | }
Mon, Apr 18 2022 2:02:59 pm | INFO: Repository finished (repository=Valuya/vattools)
Mon, Apr 18 2022 2:02:59 pm | "durationMs": 4604
Mon, Apr 18 2022 2:02:59 pm | INFO: Renovate is exiting with a non-zero code due to the following logged errors
Mon, Apr 18 2022 2:02:59 pm | "loggerErrors": [
Mon, Apr 18 2022 2:02:59 pm | {
Mon, Apr 18 2022 2:02:59 pm | "name": "renovate",
Mon, Apr 18 2022 2:02:59 pm | "level": 50,
Mon, Apr 18 2022 2:02:59 pm | "logContext": "i9jpWlDc9KN5KB5e5YKP_",
Mon, Apr 18 2022 2:02:59 pm | "repository": "charlyghislain/charlyghislaindotcom",
Mon, Apr 18 2022 2:02:59 pm | "branch": "renovate_core-js-3.x",
Mon, Apr 18 2022 2:02:59 pm | "err": {
Mon, Apr 18 2022 2:02:59 pm | "name": "HTTPError",
Mon, Apr 18 2022 2:02:59 pm | "code": "ERR_NON_2XX_3XX_RESPONSE",
Mon, Apr 18 2022 2:02:59 pm | "timings": {
Mon, Apr 18 2022 2:02:59 pm | "start": 1650283211764,
Mon, Apr 18 2022 2:02:59 pm | "socket": 1650283211764,
Mon, Apr 18 2022 2:02:59 pm | "lookup": 1650283211766,
Mon, Apr 18 2022 2:02:59 pm | "connect": 1650283211766,
Mon, Apr 18 2022 2:02:59 pm | "secureConnect": 1650283211773,
Mon, Apr 18 2022 2:02:59 pm | "upload": 1650283211773,
Mon, Apr 18 2022 2:02:59 pm | "response": 1650283211830,
Mon, Apr 18 2022 2:02:59 pm | "end": 1650283211830,
Mon, Apr 18 2022 2:02:59 pm | "phases": {
Mon, Apr 18 2022 2:02:59 pm | "wait": 0,
Mon, Apr 18 2022 2:02:59 pm | "dns": 2,
Mon, Apr 18 2022 2:02:59 pm | "tcp": 0,
Mon, Apr 18 2022 2:02:59 pm | "tls": 7,
Mon, Apr 18 2022 2:02:59 pm | "request": 0,
Mon, Apr 18 2022 2:02:59 pm | "firstByte": 57,
Mon, Apr 18 2022 2:02:59 pm | "download": 0,
Mon, Apr 18 2022 2:02:59 pm | "total": 66
Mon, Apr 18 2022 2:02:59 pm | }
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "message": "Response code 500 (Internal Server Error)",
Mon, Apr 18 2022 2:02:59 pm | "stack": "HTTPError: Response code 500 (Internal Server Error)\n at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n at processTicksAndRejections (internal/process/task_queues.js:95:5)",
Mon, Apr 18 2022 2:02:59 pm | "options": {
Mon, Apr 18 2022 2:02:59 pm | "headers": {
Mon, Apr 18 2022 2:02:59 pm | "user-agent": "RenovateBot/32.24.0 (https://github.com/renovatebot/renovate)",
Mon, Apr 18 2022 2:02:59 pm | "authorization": "***********",
Mon, Apr 18 2022 2:02:59 pm | "accept": "application/json",
Mon, Apr 18 2022 2:02:59 pm | "content-type": "application/json",
Mon, Apr 18 2022 2:02:59 pm | "content-length": "76627",
Mon, Apr 18 2022 2:02:59 pm | "accept-encoding": "gzip, deflate, br"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "url": "https://gitea.valuya.com/api/v1/repos/charlyghislain/charlyghislaindotcom/pulls/21",
Mon, Apr 18 2022 2:02:59 pm | "hostType": "gitea",
Mon, Apr 18 2022 2:02:59 pm | "username": "",
Mon, Apr 18 2022 2:02:59 pm | "password": "",
Mon, Apr 18 2022 2:02:59 pm | "method": "PATCH",
Mon, Apr 18 2022 2:02:59 pm | "http2": false
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "response": {
Mon, Apr 18 2022 2:02:59 pm | "statusCode": 500,
Mon, Apr 18 2022 2:02:59 pm | "statusMessage": "Internal Server Error",
Mon, Apr 18 2022 2:02:59 pm | "body": {"message": "", "url": "https://gitea.valuya.com/api/swagger"},
Mon, Apr 18 2022 2:02:59 pm | "headers": {
Mon, Apr 18 2022 2:02:59 pm | "date": "Mon, 18 Apr 2022 12:00:11 GMT",
Mon, Apr 18 2022 2:02:59 pm | "content-type": "application/json;charset=utf-8",
Mon, Apr 18 2022 2:02:59 pm | "content-length": "60",
Mon, Apr 18 2022 2:02:59 pm | "connection": "close",
Mon, Apr 18 2022 2:02:59 pm | "set-cookie": [
Mon, Apr 18 2022 2:02:59 pm | "i_like_gitea=eded2577b8f9d19c; Path=/; HttpOnly; SameSite=Lax",
Mon, Apr 18 2022 2:02:59 pm | "_csrf=kIOhdcYPGiN9rsPloihrd5HNzvA6MTY1MDI4MzIxMTc3ODY5NzY1Mw; Path=/; Expires=Tue, 19 Apr 2022 12:00:11 GMT; HttpOnly; SameSite=Lax"
Mon, Apr 18 2022 2:02:59 pm | ],
Mon, Apr 18 2022 2:02:59 pm | "x-content-type-options": "nosniff",
Mon, Apr 18 2022 2:02:59 pm | "x-frame-options": "SAMEORIGIN",
Mon, Apr 18 2022 2:02:59 pm | "strict-transport-security": "max-age=15724800; includeSubDomains"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "httpVersion": "1.1"
Mon, Apr 18 2022 2:02:59 pm | }
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "msg": "Failed to ensure PR: Update dependency core-js to v3"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | {
Mon, Apr 18 2022 2:02:59 pm | "name": "renovate",
Mon, Apr 18 2022 2:02:59 pm | "level": 50,
Mon, Apr 18 2022 2:02:59 pm | "logContext": "i9jpWlDc9KN5KB5e5YKP_",
Mon, Apr 18 2022 2:02:59 pm | "repository": "Valuya/comptaplan_app",
Mon, Apr 18 2022 2:02:59 pm | "branch": "renovate_googleapis-100.x",
Mon, Apr 18 2022 2:02:59 pm | "err": {
Mon, Apr 18 2022 2:02:59 pm | "name": "HTTPError",
Mon, Apr 18 2022 2:02:59 pm | "code": "ERR_NON_2XX_3XX_RESPONSE",
Mon, Apr 18 2022 2:02:59 pm | "timings": {
Mon, Apr 18 2022 2:02:59 pm | "start": 1650283273054,
Mon, Apr 18 2022 2:02:59 pm | "socket": 1650283273054,
Mon, Apr 18 2022 2:02:59 pm | "lookup": 1650283273055,
Mon, Apr 18 2022 2:02:59 pm | "connect": 1650283273055,
Mon, Apr 18 2022 2:02:59 pm | "secureConnect": 1650283273056,
Mon, Apr 18 2022 2:02:59 pm | "upload": 1650283273056,
Mon, Apr 18 2022 2:02:59 pm | "response": 1650283273153,
Mon, Apr 18 2022 2:02:59 pm | "end": 1650283273153,
Mon, Apr 18 2022 2:02:59 pm | "phases": {
Mon, Apr 18 2022 2:02:59 pm | "wait": 0,
Mon, Apr 18 2022 2:02:59 pm | "dns": 1,
Mon, Apr 18 2022 2:02:59 pm | "tcp": 0,
Mon, Apr 18 2022 2:02:59 pm | "tls": 1,
Mon, Apr 18 2022 2:02:59 pm | "request": 0,
Mon, Apr 18 2022 2:02:59 pm | "firstByte": 97,
Mon, Apr 18 2022 2:02:59 pm | "download": 0,
Mon, Apr 18 2022 2:02:59 pm | "total": 99
Mon, Apr 18 2022 2:02:59 pm | }
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "message": "Response code 500 (Internal Server Error)",
Mon, Apr 18 2022 2:02:59 pm | "stack": "HTTPError: Response code 500 (Internal Server Error)\n at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n at runMicrotasks (<anonymous>)\n at processTicksAndRejections (internal/process/task_queues.js:95:5)",
Mon, Apr 18 2022 2:02:59 pm | "options": {
Mon, Apr 18 2022 2:02:59 pm | "headers": {
Mon, Apr 18 2022 2:02:59 pm | "user-agent": "RenovateBot/32.24.0 (https://github.com/renovatebot/renovate)",
Mon, Apr 18 2022 2:02:59 pm | "authorization": "***********",
Mon, Apr 18 2022 2:02:59 pm | "accept": "application/json",
Mon, Apr 18 2022 2:02:59 pm | "content-type": "application/json",
Mon, Apr 18 2022 2:02:59 pm | "content-length": "86108",
Mon, Apr 18 2022 2:02:59 pm | "accept-encoding": "gzip, deflate, br"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "url": "https://gitea.valuya.com/api/v1/repos/Valuya/comptaplan_app/pulls/5",
Mon, Apr 18 2022 2:02:59 pm | "hostType": "gitea",
Mon, Apr 18 2022 2:02:59 pm | "username": "",
Mon, Apr 18 2022 2:02:59 pm | "password": "",
Mon, Apr 18 2022 2:02:59 pm | "method": "PATCH",
Mon, Apr 18 2022 2:02:59 pm | "http2": false
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "response": {
Mon, Apr 18 2022 2:02:59 pm | "statusCode": 500,
Mon, Apr 18 2022 2:02:59 pm | "statusMessage": "Internal Server Error",
Mon, Apr 18 2022 2:02:59 pm | "body": {"message": "", "url": "https://gitea.valuya.com/api/swagger"},
Mon, Apr 18 2022 2:02:59 pm | "headers": {
Mon, Apr 18 2022 2:02:59 pm | "date": "Mon, 18 Apr 2022 12:01:13 GMT",
Mon, Apr 18 2022 2:02:59 pm | "content-type": "application/json;charset=utf-8",
Mon, Apr 18 2022 2:02:59 pm | "content-length": "60",
Mon, Apr 18 2022 2:02:59 pm | "connection": "close",
Mon, Apr 18 2022 2:02:59 pm | "set-cookie": [
Mon, Apr 18 2022 2:02:59 pm | "i_like_gitea=ee87e2839c3e5a42; Path=/; HttpOnly; SameSite=Lax",
Mon, Apr 18 2022 2:02:59 pm | "_csrf=rS-pht4yrsvymzrVNNKKThR9S0M6MTY1MDI4MzI3MzA1NzYwMjEyMA; Path=/; Expires=Tue, 19 Apr 2022 12:01:13 GMT; HttpOnly; SameSite=Lax"
Mon, Apr 18 2022 2:02:59 pm | ],
Mon, Apr 18 2022 2:02:59 pm | "x-content-type-options": "nosniff",
Mon, Apr 18 2022 2:02:59 pm | "x-frame-options": "SAMEORIGIN",
Mon, Apr 18 2022 2:02:59 pm | "strict-transport-security": "max-age=15724800; includeSubDomains"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "httpVersion": "1.1"
Mon, Apr 18 2022 2:02:59 pm | }
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "msg": "Failed to ensure PR: Update dependency googleapis to v100"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | {
Mon, Apr 18 2022 2:02:59 pm | "name": "renovate",
Mon, Apr 18 2022 2:02:59 pm | "level": 50,
Mon, Apr 18 2022 2:02:59 pm | "logContext": "i9jpWlDc9KN5KB5e5YKP_",
Mon, Apr 18 2022 2:02:59 pm | "repository": "Valuya/embedded-webview",
Mon, Apr 18 2022 2:02:59 pm | "branch": "renovate_googleapis-100.x",
Mon, Apr 18 2022 2:02:59 pm | "err": {
Mon, Apr 18 2022 2:02:59 pm | "name": "HTTPError",
Mon, Apr 18 2022 2:02:59 pm | "code": "ERR_NON_2XX_3XX_RESPONSE",
Mon, Apr 18 2022 2:02:59 pm | "timings": {
Mon, Apr 18 2022 2:02:59 pm | "start": 1650283284392,
Mon, Apr 18 2022 2:02:59 pm | "socket": 1650283284392,
Mon, Apr 18 2022 2:02:59 pm | "lookup": 1650283284393,
Mon, Apr 18 2022 2:02:59 pm | "connect": 1650283284394,
Mon, Apr 18 2022 2:02:59 pm | "secureConnect": 1650283284397,
Mon, Apr 18 2022 2:02:59 pm | "upload": 1650283284397,
Mon, Apr 18 2022 2:02:59 pm | "response": 1650283284435,
Mon, Apr 18 2022 2:02:59 pm | "end": 1650283284435,
Mon, Apr 18 2022 2:02:59 pm | "phases": {
Mon, Apr 18 2022 2:02:59 pm | "wait": 0,
Mon, Apr 18 2022 2:02:59 pm | "dns": 1,
Mon, Apr 18 2022 2:02:59 pm | "tcp": 1,
Mon, Apr 18 2022 2:02:59 pm | "tls": 3,
Mon, Apr 18 2022 2:02:59 pm | "request": 0,
Mon, Apr 18 2022 2:02:59 pm | "firstByte": 38,
Mon, Apr 18 2022 2:02:59 pm | "download": 0,
Mon, Apr 18 2022 2:02:59 pm | "total": 43
Mon, Apr 18 2022 2:02:59 pm | }
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "message": "Response code 500 (Internal Server Error)",
Mon, Apr 18 2022 2:02:59 pm | "stack": "HTTPError: Response code 500 (Internal Server Error)\n at Request.<anonymous> (/usr/src/app/node_modules/got/dist/source/as-promise/index.js:117:42)\n at runMicrotasks (<anonymous>)\n at processTicksAndRejections (internal/process/task_queues.js:95:5)",
Mon, Apr 18 2022 2:02:59 pm | "options": {
Mon, Apr 18 2022 2:02:59 pm | "headers": {
Mon, Apr 18 2022 2:02:59 pm | "user-agent": "RenovateBot/32.24.0 (https://github.com/renovatebot/renovate)",
Mon, Apr 18 2022 2:02:59 pm | "authorization": "***********",
Mon, Apr 18 2022 2:02:59 pm | "accept": "application/json",
Mon, Apr 18 2022 2:02:59 pm | "content-type": "application/json",
Mon, Apr 18 2022 2:02:59 pm | "content-length": "86108",
Mon, Apr 18 2022 2:02:59 pm | "accept-encoding": "gzip, deflate, br"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "url": "https://gitea.valuya.com/api/v1/repos/Valuya/embedded-webview/pulls/6",
Mon, Apr 18 2022 2:02:59 pm | "hostType": "gitea",
Mon, Apr 18 2022 2:02:59 pm | "username": "",
Mon, Apr 18 2022 2:02:59 pm | "password": "",
Mon, Apr 18 2022 2:02:59 pm | "method": "PATCH",
Mon, Apr 18 2022 2:02:59 pm | "http2": false
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "response": {
Mon, Apr 18 2022 2:02:59 pm | "statusCode": 500,
Mon, Apr 18 2022 2:02:59 pm | "statusMessage": "Internal Server Error",
Mon, Apr 18 2022 2:02:59 pm | "body": {"message": "", "url": "https://gitea.valuya.com/api/swagger"},
Mon, Apr 18 2022 2:02:59 pm | "headers": {
Mon, Apr 18 2022 2:02:59 pm | "date": "Mon, 18 Apr 2022 12:01:24 GMT",
Mon, Apr 18 2022 2:02:59 pm | "content-type": "application/json;charset=utf-8",
Mon, Apr 18 2022 2:02:59 pm | "content-length": "60",
Mon, Apr 18 2022 2:02:59 pm | "connection": "close",
Mon, Apr 18 2022 2:02:59 pm | "set-cookie": [
Mon, Apr 18 2022 2:02:59 pm | "i_like_gitea=ab66209a94debd16; Path=/; HttpOnly; SameSite=Lax",
Mon, Apr 18 2022 2:02:59 pm | "_csrf=-LxEt48W_POQyGGNzUBkbvM96yE6MTY1MDI4MzI4NDM5OTA1Mzg2MA; Path=/; Expires=Tue, 19 Apr 2022 12:01:24 GMT; HttpOnly; SameSite=Lax"
Mon, Apr 18 2022 2:02:59 pm | ],
Mon, Apr 18 2022 2:02:59 pm | "x-content-type-options": "nosniff",
Mon, Apr 18 2022 2:02:59 pm | "x-frame-options": "SAMEORIGIN",
Mon, Apr 18 2022 2:02:59 pm | "strict-transport-security": "max-age=15724800; includeSubDomains"
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "httpVersion": "1.1"
Mon, Apr 18 2022 2:02:59 pm | }
Mon, Apr 18 2022 2:02:59 pm | },
Mon, Apr 18 2022 2:02:59 pm | "msg": "Failed to ensure PR: Update dependency googleapis to v100"
Mon, Apr 18 2022 2:02:59 pm | }
Mon, Apr 18 2022 2:02:59 pm | ]

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
A 500 would indicate that there's a bug in Gitea, but potentially we can work around it by reducing our maximum PR body length to less than what its documented capability is.

**Comment by cghislai** [comment]
> A 500 would indicate that there's a bug in Gitea, but potentially we can work around it by reducing our maximum PR body length to less than what its documented capability is.

Yes, especially since it appears to abort the bot job (edit: i have to recheck, not sure about this rn). Gracefully handling platform errors on a per-repo basis would be nice.

**Comment by rarkins** [comment]
I think the importance of this depends on how it impacts Renovate's future processing. If it just impacts the current PR, it's lowest impact. If it stops the entire job, then it's quite high.

**Comment by cghislai** [comment]
@rarkins my bad, the error does not abort the job, i was mistaken, sorry. 

However, the kubernetes job is marked as failed, which cause another pod to be launched, (until it reaches the job backoff limit: 6 attemps). 

I think the current behaviour is acceptable, although a sucessful job that would have reported the error on the pr could be better. Feel free to close.

For reference, I created an issue on the gitea repo as well: https://github.com/go-gitea/gitea/issues/19471

**Comment by viceice** [comment, marked as answer]
it was a bug 🐛 in gitea, which is already resolved in a later version.

- https://github.com/go-gitea/gitea/issues/19471
